incubator/orm/indexer.go

```diff
@@ -0,0 +1,119 @@
+package orm
+
+import (
+	"encoding/binary"
+	"math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// IndexerFunc creates one or multiple index keys for the source object.
+type IndexerFunc func(value interface{}) ([][]byte, error)
+
+// Indexer manages the persistence for an index based on searchable keys and operations.
+type Indexer struct {
+	indexerFunc IndexerFunc
+	addPolicy   func(store sdk.KVStore, secondaryIndexKey []byte, rowId uint64) error
```

Comment:
I like this design. One if statement in the constructor to unify what could be two classes